### ISSUE TITLE
/new experiment to add FxA CTA 

### ISSUE THREAD

**Issue by thejeffreytaylor** [mozilla/bedrock#5974]
[Experiment overview deck](https://docs.google.com/presentation/d/109QM--B0e1bndEov7wpCJWbvHmy26Hh-Wj6KOxObjmY/edit?usp=sharing). 

Success Criteria:
- [x] Uses the FxAccount iframeless form
- [ ] Metrics must track impact of FxA CTA on downloads
- [x] Experiment content is shown only to up-to-date Firefox users
- [x] Fx Accounts team is informed of launch, in advance of it

**Comment by alexgibson** [comment]
This task is somewhat dependent on the status of https://github.com/mozilla/bedrock/pull/5713

**Comment by thejeffreytaylor** [comment]
And is a sibling to #5944 - Once all these experiments are done, we need to make sure that our email collection efforts don't confuse users (Fxa vs. onboarding, etc.), and that the flows work such that users don't feel pestered or confused by the timing of these asks.

**Comment by dzingeek** [comment]
Here is a .png of the user flow. @mrkwvr is working on the ui for the page now.

![new_fxa-experiment_uf](https://user-images.githubusercontent.com/3740746/44061489-6f66ba06-9f0d-11e8-85d5-4f96ab874cf3.png)

**Comment by ejregithub** [comment]
Eric to ask jpetto for status on PR for iframeless

**Comment by ejregithub** [comment]
I pinged Alex Davis about this two weeks ago, he's good with it.

**Comment by ejregithub** [comment]
@stephaniehobson - as a prerequisite to this experiment running, could you please review @jpetto 's PR for the iframeless FxAccount form? https://github.com/mozilla/bedrock/pull/5713

**Comment by mrkwvr** [comment]
Final design file https://mozilla.box.com/s/y3gjvsl7d1ny1a2tm0a3wbr1fib5l1q4

**Comment by dzingeek** [comment]
Looks like we got approval from Lauren to move forward with this direction and copy via slack. 
08/23/18 @ 12:31pm

**Comment by stephaniehobson** [comment]
#5713 is merged now.

The original requirements for the iframeless form were to put it on /firstrun which is only ever seen by Firefox users.

So, even though #5713 is merged it is not ready to be used on /new immediately.

**Comment by stephaniehobson** [comment]
As @thejeffreytaylor pointed out, we now ask the users for email addresses on both the download and the download thanks pages with virtually identical forms.

<img width="1257" alt="screen shot 2018-09-04 at 4 02 04 pm" src="https://user-images.githubusercontent.com/854701/45062031-fd1f4e80-b05b-11e8-97cd-bed0d933390f.png">

<img width="1276" alt="screen shot 2018-09-04 at 4 00 40 pm" src="https://user-images.githubusercontent.com/854701/45062036-01e40280-b05c-11e8-8b48-38ac55ce5421.png">

@dzingeek Do you think we should go forward with this test?

**Comment by stephaniehobson** [comment]
Oh, it looks like the mobile download scene2 is still in the code... I don't know if we can forward an en-US user to it but I will see.

**Comment by stephaniehobson** [comment]
#6092 is changing the default "Thank you" page and we don't currently have a way to send variants to a different thank you page. We have it for experiences though so something should be possible with more work.

I'll talk to @alexgibson about what he's planning for #5999, which will have the same problem.

**Comment by ejregithub** [comment]
Blocked on other work

**Comment by stephaniehobson** [comment]
#6092 is merged and Alex gave me the info I needed from #5999 so this is unblocked.

The iframeless form has slightly different content than the Mockup did. @dzingeek Are you okay with this: https://www-demo4.allizom.org/en-US/firefox/new/?v=x And if so, can we run it past the client too.

**Comment by stephaniehobson** [comment]
(Also, I left the browser product shot to minimize the number of variables we are testing. If that's not okay let me know)

**Comment by dzingeek** [comment]
@stephaniehobson a few things

- [ ] "Enter your email to access Firefox Accounts" should all be the same font size and weight.
- [ ] Move the "By proceeding, you agree..." copy below the Continue button.
- [ ] Remove the image of the screenshot
- [ ] Add the Refresh Firefox button [(see design mockup ^)](https://github.com/mozilla/bedrock/issues/5974#issuecomment-418545540)

<img width="1148" alt="screen shot 2018-09-12 at 3 34 59 pm" src="https://user-images.githubusercontent.com/3740746/45456916-785ab300-b6a1-11e8-8e2a-604faa75c5f4.png">

**Comment by stephaniehobson** [comment]
I'm not sure who we need to talk to about why the text in the form is where it is and whether or not we can move it. @jpetto do you have any context?

@dzingeek  The refresh button should appear if you are eligible to refresh. Did you view the page on Release?

**Comment by craigcook** [comment]
I updated the form macro and put the button right after the field, and the agreement after that, but I think that's wrong and wanted to bring it up. 

The disclaimer really should come before the button so it reads properly, especially since it's phrased "by proceeding..." If it comes after the button you will have already submitted it before you can accept the terms. Logically the agreement needs to occur before the button in the source order but we may be able to do some code and layout tricks to make everyone happy.

**Comment by stephaniehobson** [comment]
Need a slight change to the user flow because of technical constraints.

To check the status of FxAccounts takes a not-small bit of JavaScript and adds half a second to the page load time. I'm moving that check to *after* the user is opted into the experiment. As a result, the signed in users will get the FxA account flow. I will check their FxA status on that page and hide the form if they're signed in. I'll post a screen shot once I've updated the flow.

**Comment by stephaniehobson** [comment]
Users on this page are segmented automatically by FxA status so we can still compare signed out users to signed out users when looking at the test results.

**Comment by alexgibson** [comment]
> If we want to make this experiment permanent we can explore using the original user flow.

If we decide we want to make this new state for FF users a default, I think we may need to rethink the design a little here. The FxA form should really be incorporated into the existing /new/ page, as opposed to redirecting to a different template altogether. For the purposes of a test this if fine and makes total sense, but if there are things we don't like about the current design then we should consider redesigning the whole page longer-term, as opposed to just individual states.

**Comment by stephaniehobson** [comment]
The switch hasn't been activated in www-config yet so it's not on.

Also, I feel like we should have a way to track that this needs to be turned off and have the data analyzed, but maybe a separate card would be better for that?

**Comment by ejregithub** [comment]
@stephaniehobson - Analytics Pros is going to QA the page and respond here.  They'll set up a dashboard for reporting as well. Target completion for QA is 10/12.

**Comment by rbwinters** [comment]
@stephaniehobson  - Following up to confirm the QA results.

1. The first item in the QA failed. There is no "data-button-name" attribute in this element. 
2. The download is returning as an incomplete pass. CD28: download language is not being passed because it's missing the data attribute in the clickable element 'data-download-language' 

The QA document is available in the project tracker. Please let us know if you have any questions.

Robert

**Comment by stephaniehobson** [comment]
@rbwinters Could you please link to the project tracker (I'm not sure what you mean by project tracker).

**Comment by rbwinters** [comment]
@stephaniehobson here you go, let me know if you need anything additional: https://docs.google.com/spreadsheets/d/1pcW4XgcNo4_kr7i9CKo5P-uMRoeve02X0BnhKKCBwLw/edit?usp=sharing

**Comment by stephaniehobson** [comment]
I updated the Continue button to have the correct attribute.

I'm not comfortable adding download-language without looking at it more carefully. That button is the same code that is used on every single download button on the site. Right now we're only recording the language on [the page that has different languages for downloads](https://www.mozilla.org/en-US/firefox/all/). If this QA review has turned up a problem with that button we should look at it carefully before making changes. 

This experiment is en-US only so I don't think language should be a blocker for this experiment.

**Comment by stephaniehobson** [comment]
Experiment was enabled in https://github.com/mozmeao/www-config/pull/149 :)

**Comment by rbwinters** [comment]
@stephaniehobson the dashboard has been created for this ticket. Let us know if you need anything additional: https://datastudio.google.com/open/1cmZ9Le5k5Ym_sF2n-XXQ0u6OQ7XPwu1D
